### PR TITLE
Provision with Packer

### PR DESCRIPTION
**Pull request by reedlaw**
This adds initial support for provisioning on multiple providers. There is initial support for EC2 and DigitalOcean.

### CODE REVIEW

**Comment by leonelgalan** [comment]
Hey Reed, but this isn't using tape at all? In #19 I was envisioning using tape to go further on the provisioning and install those common things that take time. So the app specific provisioning could be shorter.

I don't know if packer is the right tool for this job, it seems you simply need to run this on every knew machine, maybe a wrapper of Digital Ocean's and EC2's API that create boxes and run the script could work better.

File: README.md [hunk 1]
```diff
@@ -2,9 +2,11 @@
 # Infrastructure Management
 
 ## Deploying & provisioning with tape
-**Use Unbuntu precise64 (12.04 x64)** 
+**Use Ubuntu Server 14.04 LTS x64**
 
-**Enable ssh access via root user**
+**Provision with [packer](https://packer.io/)**
+
+```packer build -var 'aws_access_key=XXXXX' -var 'aws_secret_key=XXXXX' -var 'do_api_key=XXXXX' -var 'do_client_id=XXXXX' packer.json```
```
Comment (leonelgalan) [inline README.md]: We need `do_api_token`, and not `do_api_key` or `do_client_id`

File: packer.json [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      "region": "us-east-1",
+      "source_ami": "ami-9a562df2",
+      "instance_type": "m3.medium",
+      "ssh_username": "ubuntu",
+      "ami_name": "packer-image {{timestamp}}"
+    },
+    {
+      "type": "digitalocean",
+      "api_token": "{{user `do_api_token`}}"
+    }]
```
Comment (leonelgalan) [inline packer.json]: The default size is 512, probably too small for our needs.

File: support/do.sh [hunk 3]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+useradd --create-home --shell /bin/bash --user-group deployer
+cp -r /root/.ssh /home/deployer
+chown -R deployer:deployer /home/deployer/.ssh 
+echo 'deployer ALL=(ALL:ALL) ALL' >> /etc/sudoers
```
Comment (leonelgalan) [inline support/do.sh]: We can't have passwordless sudo for deployer.

**Comment by leonelgalan** [comment]
Why not set the user at the inventory: `ansible_ssh_user=ubuntu` for EC2 machines?

File: README.md (hunk 1 above)
Comment (reedlaw) [inline README.md]: You're right. It should be `do_api_token` for the latest authentication method, but I get validation errors from packer. Not sure if this is fixed but I don't use DO for any projects so I don't have much time to test it.

File: packer.json (hunk 2 above)
Comment (reedlaw) [inline packer.json]: Like I said above, none of my work projects are on DO. I need someone else to take over support for DO.

File: support/do.sh (hunk 3 above)
Comment (reedlaw) [inline support/do.sh]: What do you propose then? Should we have a `provisioner` user with sudo privedges? Or should we whitelist the commands available to `deployer`?

**Comment by reedlaw** [comment]
@leonelgalan that's fine but what this PR is really meant to address is the current `root` requirement. I don't care what user we use as long as it isn't root. And if it's not a default user across all hosting providers it should be added automatically.

File: support/do.sh (hunk 3 above)
Comment (leonelgalan) [inline support/do.sh]: Why not root? I'm sure we could use a shell provisioner to enable root on ubuntu. Every other distro should have a root user right?

Comment (BrandonMathis) [inline support/do.sh]: I'm wiht @leonelgalan on this one. Lets make the provisioner server root.

**Comment by BrandonMathis** [comment]
Using root to provision a box seems like pretty reasonable behavior @reedlaw. During provisioning, we have to be logged in as a user with top-level access and I prefer to use the industry standard of 'root'

**Comment by reedlaw** [comment]
@BrandonMathis, most official Amazon AMIs don't enable root access. Ansible allows for specifying which tasks are run with `sudo` so that the entire playbook doesn't need to be run with root. If we must run as root then the process for enabling root should be automated. I'm not sure what the plugin architecture for tape is supposed to be and I'm running out of time again so I'm not sure if I'll have the chance.

**Pull request closed without merging**